lib/bundler/ruby_version.rb

```diff
@@ -100,6 +100,19 @@ def self.system
       @ruby_version ||= RubyVersion.new(RUBY_VERSION.dup, RUBY_PATCHLEVEL.to_s, ruby_engine, ruby_engine_version)
     end
 
+    def gem_version_with_patch
+      unless @gem_version_with_patch
+        @gem_version_with_patch = Gem::Requirement.create("#{@gem_version}.0").requirements.first.last
+        if @patchlevel
+          begin
```

Comment:
it's meant to handle the `BadRequirementError` exception. But I'm not really sure this even works for that case. I found a better way to do this already